README.rst

```diff
@@ -0,0 +1,670 @@
+Tutorial - Fixing ROS 2 ASAN/TSAN Issues
+========================================
+
+Note: this tutorial is mainly focused around ASAN but TSAN issues can be
+handled in similar way.
+
+Pre-Requisites
+--------------
+
+-  Install all ROS 2 build dependencies using the B9 script or following
+   the `official
+   instructions <https://index.ros.org/doc/ros2/Installation/Linux-Development-Setup/>`__.
+-  Install and enable Colcon mixin package:
+
+::
+
+    sudo apt-get install python3-colcon-mixin
+    colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
+    colcon mixin update default
+
+Mixins are “configuration profiles” which can be used to compile ROS 2
+with additional flags more easily. They are stored locally in
+``~/.colcon``. For instance, you can see the `asan mixin configuration
+file
+here <https://github.com/colcon/colcon-mixin-repository/blob/master/asan.mixin>`__.
+
+-  Setup ``ccache`` to speed-up development:
+
+::
+
+    sudo apt install -y ccache
+    ccache -M 20G # increase cache size
+    # Add the following to your .bashrc or .zshrc file and restart your terminal:
+    export CC=/usr/lib/ccache/gcc
+    export CXX=/usr/lib/ccache/g++
+
+-  Create a ROS 2 workspace for ASAN related tasks (do not compile yet!)
+
+::
+
+    cd  # ros2_asan_ws can be changed by any other string
+    cd ~/ros2_asan_ws
+    wget https://raw.githubusercontent.com/ros2/ros2/master/ros2.repos
+    vcs-import src < ros2.repos
+
+Fixing your workspace
+---------------------
+
+Some packages in ROS 2 do not compile properly and should be ignored:
+
+::
+
+    touch src/ros2/common_interfaces/actionlib_msgs/COLCON_IGNORE
+    touch src/ros2/common_interfaces/common_interfaces/COLCON_IGNORE
+    touch src/ros2/rosidl_typesupport_opensplice/opensplice_cmake_module/COLCON_IGNORE
+    touch src/ros2/rmw_fastrtps/rmw_fastrtps_dynamic_cpp/COLCON_IGNORE
+    touch src/ros2/rmw_opensplice/rmw_opensplice_cpp/COLCON_IGNORE
+    touch src/ros2/ros1_bridge/COLCON_IGNORE
+    touch src/ros2/rosidl_typesupport_opensplice/rosidl_typesupport_opensplice_c/COLCON_IGNORE
+    touch src/ros2/rosidl_typesupport_opensplice/rosidl_typesupport_opensplice_cpp/COLCON_IGNORE
+    touch src/ros2/common_interfaces/shape_msgs/COLCON_IGNORE
+    touch src/ros2/common_interfaces/stereo_msgs/COLCON_IGNORE
+    touch src/ros2/common_interfaces/trajectory_msgs/COLCON_IGNORE
```

Comment:
It might be too much to ask but can you elaborate a bit why each of these don't compile properly?

Reply:
@thomas-moulard might have the answer to this one. Otherwise I'll try compiling these and see what happens.

Reply:
@dirk-thomas This is what the CI is currently doing (last time I checked) so we never want to try more than what the CI is attempting to do.

Reply:
Some of the packages are only ignored to save time so the current description might be misleading.

Reply:
Turns out I've not been following this part of the tutorial and haven't had problems. I'll delete it unless there are objections.